x-pack/plugins/observability_solution/observability_onboarding/server/routes/flow/route.ts

```diff
@@ -335,10 +338,21 @@ const integrationsInstallRoute = createObservabilityOnboardingServerRoute({
 
     let installedIntegrations: InstalledIntegration[] = [];
     try {
-      installedIntegrations = await ensureInstalledIntegrations(
+      const settledResults = await ensureInstalledIntegrations(
         integrationsToInstall,
         packageClient
       );
+      installedIntegrations = settledResults.reduce<InstalledIntegration[]>((acc, result) => {
```

Comment:
Unfortunately not. It could be replaced with a `filter` followed by a `map` but then I'd need to iterate through the array twice and it wouldn't be exactly simpler:

```ts
      installedIntegrations = settledResults
        .filter(
          (result): result is PromiseFulfilledResult<InstalledIntegration> =>
            result.status === 'fulfilled'
        )
        .map((result) => result.value);
```

Reply:
ah, true 👍